server/src/main/java/io/crate/analyze/relations/PlannedRelation.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.analyze.relations;
+
+import static io.crate.common.collections.Iterables.getFirst;
+
+import java.util.List;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import io.crate.exceptions.AmbiguousColumnException;
+import io.crate.exceptions.ColumnUnknownException;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.table.Operation;
+import io.crate.planner.operators.LogicalPlan;
+
+public class PlannedRelation implements AnalyzedRelation {
+
+    private LogicalPlan source;
+
+    public PlannedRelation(LogicalPlan source) {
+        this.source = source;
+    }
+
+    @Override
+    public <C, R> R accept(AnalyzedRelationVisitor<C, R> visitor, C context) {
+        return visitor.visitPlannedRelation(this, context);
+    }
+
+    @Override
+    public @Nullable Symbol getField(
+        ColumnIdent column,
+        Operation operation,
+        boolean errorOnUnknownObjectKey) throws AmbiguousColumnException, ColumnUnknownException, UnsupportedOperationException {
+        throw new UnsupportedOperationException("Not supported on PlannedRelation");
+    }
+
+    @Override
+    public RelationName relationName() {
+        return getFirst(source.relationNames(), null);
+    }
+
+    @Override
+    public @NotNull List<Symbol> outputs() {
+        return source.outputs();
+    }
+
+    @Override
+    public String toString() {
+        return source.relationNames().getFirst().toString();
```

Comment:
What would be a good printable representation here? Show the full plan of the source ?

Reply:
I think this is fine as is, but could use `relationName().toString()`